gapic/src/main/com/google/gapid/perfetto/models/Selection.java

```diff
@@ -133,17 +145,19 @@ public void addSelection(MultiSelection other) {
     }
 
     public void markTime(State state) {
-      // TODO: is this good enough?
-      if (selections.size() == 1) {
-        firstSelection().markTime(state);
-      }
+      getRange().ifNotEmpty(state::setHighlight);
     }
 
     public void zoom(State state) {
-      // TODO: handle zooming into multiple selections.
-      if (selections.size() == 1) {
-        firstSelection().zoom(state);
+      getRange().ifNotEmpty(state::setVisibleTime);
+    }
+
+    private TimeSpan getRange() {
+      TimeSpan[] range = new TimeSpan[] { TimeSpan.ZERO };
```

Comment:
Can we use `new TimeSpan(0, 0)` here instead of an array?

Reply:
no, because it has to be final. Putting it into an array allows us to modify it.

Reply:
Thanks for explaining!